### ISSUE TITLE
Vehicles entering and stopping in the middle of intersections

### ISSUE THREAD

**Issue by oshadajay** [eclipse-sumo/sumo#14927]
Hi, if you pay attention to the following video, vehicles keep entering the intersection even when they can't pass the intersection and that results in a traffic jam. Ultimately that results in multiple vehicles teleporting. In addition to that a vehicle gets stuck in the middle of the intersection for no apparent reason when it has space to enter. Ideally, a vehicle should only enter an intersection if it have enough space in the road it is entering into and intersections must be kept clear. Is there a way to resolve this? Thank you.

https://github.com/eclipse-sumo/sumo/assets/55861474/64acb430-5a79-401c-8a49-89776a66985f

**SUMO-version:**
1.20

**operating system:**
Windows

**Comment by namdre** [comment]
SUMO already has plenty of code that intends to prevent "entering an intersection if the downstream end is jammed". This code checks for stopped vehicles though. From the look of your video it seems that the problem is caused by slow moving downstream vehicles which then fail to trigger the downstream-jam detection.

**Comment by oshadajay** [comment]
Yeah seems like so. And the vehicles parked at the downstream end could also affect the situation adversely. Appreciate your support on this. Thank you.
